modules/standard/IO.chpl

```diff
@@ -2208,6 +2209,27 @@ proc file.getPath(out error:syserr) : string {
 
 /*
 
+Returns the parent directory of the :type:`~IO.file` record.  For instance,
+a file with path `/foo/bar/baz` would return `/foo/bar`
+
+ :return: The parent directory of the file
+ :rtype: `string`
+
+*/
+proc file.getParentName(out error:syserr): string {
+  check();
+  var ret:string;
+  on this.home {
+    var tmp: c_string_copy;
+    error = qio_file_path(_file_internal, tmp);
```

Comment:
Since much of this is similar to file.getPath() above, couldn't you just call that to get the string you need to send to dirname?  Or file.realPath()?  (Not sure which is better in this particular situation)